src/actions/websocket.js

```diff
@@ -582,12 +580,17 @@ function handleChannelUpdatedEvent(msg, dispatch, getState) {
     }
 }
 
-function handleChannelConvertedEvent(msg, dispatch) {
-    try {
-        const channelId = msg.data ? JSON.parse(msg.data.channel_id) : null;
-        dispatch(getChannel(channelId));
-    } catch (err) {
-        return;
+// handleChannelConvertedEvent handles updating of channel which is converted from public to private
+function handleChannelConvertedEvent(msg, dispatch, getState) {
+    const channelId = msg.data.channel_id;
+    if (channelId) {
```

Comment:
what happens if you don't have the channel in the state? shouldn't you get the channel from the server?

Reply:
No need to get since the user is either not a channel member or have not attempted to open the MoreChannels modal (which lists joinable channels).